marklogic-data-hub/src/test/java/com/marklogic/hub/dhs/DeployUserAmpsTest.java

```diff
@@ -0,0 +1,98 @@
+package com.marklogic.hub.dhs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.marklogic.appdeployer.command.security.DeployAmpsCommand;
+import com.marklogic.hub.AbstractHubCoreTest;
+import com.marklogic.hub.HubConfig;
+import com.marklogic.hub.impl.Versions;
+import com.marklogic.mgmt.resource.security.AmpManager;
+import com.marklogic.rest.util.ResourcesFragment;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import java.io.IOException;
+import java.nio.file.Path;
+
+
+public class DeployUserAmpsTest extends AbstractHubCoreTest {
+    ObjectMapper mapper = new ObjectMapper();
+
+    @BeforeEach
+    void checkIfTestsCanBeRun(){
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assertions.assertTrue(mlVersion.isNightly() || (mlVersion.getMajor() == 10 && mlVersion.getMinor() >= 440));
+    }
+
+    @Test
+    void createAmpWithInheritableRole() {
+        //POST request
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME, "data-hub-developer");
+        ResourcesFragment amps = new AmpManager(runAsDataHubSecurityAdmin().getManageClient()).getAsXml();
+        Assertions.assertTrue(amps.resourceExists("testAmp"));
+
+        //PUT request(doesn't work)
+        writeAmpFileToProjectAndDeploy(HubConfig.DEFAULT_MODULES_DB_NAME,"data-hub-developer", "data-hub-operator");
```

Comment:
What error do you get for this?

Reply:
```
11:32:45.559 [main] ERROR c.m.r.util.MgmtResponseErrorHandler - Logging HTTP response body to assist with debugging: {"errorResponse":{"statusCode":"403", "status":"Forbidden", "messageCode":"", "message":"You do not have credentials to access /manage/v2/amps/testAmp/properties?document-uri=/data-hub/5/impl/jobs.sjs&modules-database=data-hub-MODULES ."}}
11:32:45.560 [main] ERROR c.m.m.resource.security.AmpManager - Error occurred while sending PUT request to /manage/v2/amps/testAmp/properties?document-uri=/data-hub/5/impl/jobs.sjs&modules-database=data-hub-MODULES; logging request body to assist with debugging: {
  "local-name" : "testAmp",
  "document-uri" : "/data-hub/5/impl/jobs.sjs",
  "modules-database" : "data-hub-MODULES",
  "role" : [ "data-hub-developer", "data-hub-operator" ]
}
```

Reply:
This may be like query rolesets, where - despite what the docs say (https://docs.marklogic.com/REST/PUT/manage/v2/amps/[id-or-name]/properties) - updates are no longer allowed.

If that's the case, then we'll need to subclass DeployAmpsCommand and modify it to not do updates. Can you see what happens if it does a POST and the amp already exists?

Reply:
I get this if I a POST request is made and the amp already exists
```
curl -X POST  --anyauth -u test-data-hub-security-admin:password -H "Content-Type:application/json" \
-d @testAmp.json http://localhost:8002/manage/v2/amps
{"errorResponse":{"statusCode":"400", "status":"Bad Request", "messageCode":"MANAGE-CONFLICTINGCONFIG", "message":"MANAGE-CONFLICTINGCONFIG: (err:FOER0000) A conflict has been detected: roles differ."}}%
```